injectproxy/routes.go

```diff
@@ -264,6 +275,9 @@ func (r *routes) query(w http.ResponseWriter, req *http.Request) {
 	// enforce in both places.
 	q, found1, err := enforceQueryValues(e, req.URL.Query())
 	if err != nil {
+		if err == ErrIllegalLabelMatcher {
+			http.Error(w, err.Error(), http.StatusForbidden)
```

Comment:
I wonder if this should rather be `StatusBadRequest`

Reply:
Yeah fair point. Since we don't actually check if the user is authorized or not `StatusBadRequest` seems more appropriate.